Metric exporting endpoint

jleeh opened the pull request:
Replaces #497 and for [#159707176](https://www.pivotaltracker.com/story/show/159707176)

`/v2/metric` endpoint that outputs metrics for all the job specs.

jleeh commented:
This is ready for review now. Endpoint is now auth'd with a bearer token that can be specified with `METRICS_BEARER_TOKEN` env var that is random if left unset.

se3000 commented on web/metrics_controller_test.go:
```diff
@@ -0,0 +1,113 @@
+	assert.Equal(t, stats.JobSpecCounts[0].StatusCount["completed"], 1, "Should have a single completed run")
+	assert.Equal(t, stats.JobSpecCounts[0].AdaptorCount["noop"], 1, "Should have noop as an adaptor")
+	assert.Equal(t, stats.JobSpecCounts[0].ParamCount["url"][0].Value, "https://chain.link", "Should include the same URL")
+	assert.Equal(t, stats.JobSpecCounts[0].ParamCount["url"][0].Count, 1, "Should include a url")
```
A lot of this seems like testing of logic that is already tested in the `service` package. You could get away with just testing some minimal piece of it here, focus on the controller specific parts here, and then you don't have to update it in two places when you make a change.

se3000 commented on web/metrics_controller.go:
```diff
@@ -0,0 +1,76 @@
+)
+
+// MetricsController fetches high-level information about the nodes usage
+type MetricsController struct {
```
Can you clarify the difference between PromController and MetricsController? I get that one is in Prometheus format, and one is not. Do you plan on using both controllers or just one?

jleeh commented:
Thanks for the comments @se3000. So based on what we discussed, I've removed the direct Prometheus integration, changed all to camelCase and cleaned up the struct names.

With removing Prometheus, I've also kept the endpoint as `/metrics` put it back behind normal session auth.

jleeh commented:
Closing this due to the re-work to use expvar that is WIP.
